### PR TITLE
Base server implementation, Django integration example

### LINKED ISSUE #87: Extract logic from WSGI middleware into reusable functions
A bulk of logic living in `GraphQLMiddleware`could be extracted and reused in server implementations like `django_ariadne`.

### PR DESCRIPTION
**Pull request by rafalp**
This PR adds reusable boilerplate for synchronous GraphQL servers, making it much simpler to implement custom servers or integrations with webframeworks.

Fixes #87, #91

### CODE REVIEW

**Comment by jrast** [comment]
If this will be merged, the example for flask should also be updated. See https://github.com/mirumee/ariadne/pull/126

I'll update the PR for Flask, remind me if I forget about it.

File: tests/test_graphql.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+
+class AlwaysInvalid(ValidationRule):
+    def leave_operation_definition(  # pylint: disable=unused-argument
+        self, *args, **kwargs
+    ):
+        self.context.report_error(GraphQLError("Invalid"))
+
+
+def test_graphql_sync_works(schema):
```
Comment (rafalp) [inline tests/test_graphql.py]:
```suggestion
def test_graphql_sync_executes_query(schema):
```

File: tests/test_graphql.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+def test_graphql_sync_uses_validation_rules(schema):
+    success, result = graphql_sync(
+        schema, {"query": '{ hello(name: "world") }'}, validation_rules=[AlwaysInvalid]
+    )
+    assert not success
+    assert result["errors"][0]["message"] == "Invalid"
+
+
+@pytest.mark.asyncio
+async def test_graphql_works(schema):
```
Comment (rafalp) [inline tests/test_graphql.py]:
```suggestion
def test_graphql_executes_query(schema):
```

Comment (patrys) [reply tests/test_graphql.py]: This has to be a coroutine.

Comment (rafalp) [reply tests/test_graphql.py]: ah, right. I've just copied over previous comment and edited.

**Pull request merged**